worker/src/lib.rs

```diff
@@ -149,22 +157,33 @@ impl Response {
         Ok(Self {
             body,
             status_code: 200,
+            edge_response: None,
         })
     }
 
     pub fn empty() -> Result<Self> {
         Ok(Self {
             body: None,
             status_code: 200,
+            edge_response: None,
         })
     }
 
     pub fn error(msg: String, status: u16) -> Result<Self> {
         Ok(Self {
             body: Some(msg),
             status_code: status,
+            edge_response: None,
         })
     }
+
+    pub fn status_code(&self) -> u16 {
+        self.status_code
+    }
+
+    pub fn edge_response(&self) -> Option<EdgeResponse> {
+        self.edge_response.clone()
```

Comment:
This should return a reference to the edge response (i.e. `Option<&EdgeResponse>`) instead of cloning it.

Reply:
good point, and done.